specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2023-01-15-preview/sqlManagedInstances.json

```diff
@@ -0,0 +1,307 @@
+      "type": "object",
+      "properties": {
+        "dataControllerId": {
+          "type": "string",
+          "description": "null"
+        },
+        "admin": {
+          "type": "string",
+          "description": "The instance admin user"
+        },
+        "startTime": {
+          "type": "string",
+          "description": "The instance start time"
+        },
+        "endTime": {
+          "type": "string",
+          "description": "The instance end time"
+        },
+        "k8sRaw": {
+          "$ref": "#/definitions/SqlManagedInstanceK8sRaw",
+          "description": "The raw kubernetes information"
+        },
+        "basicLoginInformation": {
+          "$ref": "./common.json#/definitions/BasicLoginInformation",
+          "description": "Username and password for basic authentication."
+        },
+        "lastUploadedDate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Last uploaded date from Kubernetes cluster. Defaults to current date time"
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The provisioning state of the Arc-enabled SQL Managed Instance resource."
+        },
+        "activeDirectoryInformation": {
+          "description": "Active Directory information related to this SQL Managed Instance.",
+          "$ref": "#/definitions/ActiveDirectoryInformation"
+        },
+        "licenseType": {
+          "type": "string",
+          "enum": [
+            "BasePrice",
+            "LicenseIncluded",
+            "DisasterRecovery"
+          ],
+          "x-ms-enum": {
+            "name": "arcSqlManagedInstanceLicenseType",
+            "modelAsString": true
+          },
+          "default": "BasePrice",
+          "description": "The license type to apply for this managed instance."
+        },
+        "clusterId": {
+          "type": "string",
+          "description": "If a CustomLocation is provided, this contains the ARM id of the connected cluster the custom location belongs to."
+        },
+        "extensionId": {
+          "type": "string",
+          "description": "If a CustomLocation is provided, this contains the ARM id of the extension the custom location belongs to."
+        }
+      }
+    },
+    "SqlManagedInstanceUpdate": {
+      "description": "An update to a SQL Managed Instance.",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SqlManagedInstanceK8sRaw": {
+      "type": "object",
+      "description": "The raw kubernetes information.",
+      "properties": {
+        "spec": {
+          "$ref": "#/definitions/SqlManagedInstanceK8sSpec"
+        }
+      },
+      "additionalProperties": {}
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=7&length=20)

Please provide a justification for why you need this at all?

Reply:
Same comment for the usage of additionalProperties in the rest of the file

Reply:
These properties will be synced from the cluster and since customers an add key value pairs, we thought this would allow us to sync those properties to azure. Also, we want to keep this behavior consistent for sql managed instances since this is how it was configured in previous API versions. @moadus would you be able to provide more details for this?